src/components/AddItemForm/AddItemForm.jsx

```diff
@@ -0,0 +1,267 @@
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useState } from 'react';
+import { Input, TextArea, SubmitButton, ListBox, ComboBox } from '../Forms';
+
+const schema = yup.object().shape({
+  title: yup.string().required('Please insert your Item Name.'),
+  price: yup.number().positive('Please insert a positive number.'),
+  description: yup.string().required('Please add a description.'),
+  email: yup
+    .string()
+    .email('Should be a valid email.')
+    .required('Please insert your Email.'),
+  country: yup.string().required('Please insert your Country name.'),
+  city: yup.string().required('Please insert your City name.'),
+  file: yup
+    .mixed()
+    .test('required', 'Please upload a photo.', (photo) => {
+      if (!photo.length) return false;
+      return true;
+    })
+    .test('fileSize', 'File size is too large.', (photo) => {
+      return photo.length && photo[0].size <= 10000000;
+    })
+    .test('fileType', 'Please add a supported file type.', (photo) => {
+      return (
+        photo.length &&
+        ['image/jpeg', 'image/png', 'image/jpg'].includes(photo[0].type)
+      );
+    }),
+});
+
+const ITEM_TYPES = ['Crafted', 'Donated', 'New'];
+
+const ITEM_CATEGORY = [
+  'Men',
+  'Women',
+  'Kids',
+  'Toys',
+  'Books',
+  'Sports',
+  'Clothes',
+  'Furniture',
+  'Electronics',
+  'Homemade',
+  'Other',
+];
+
+const INTIAL_VALUES = {
+  title: '',
+  price: 0,
+  description: '',
+  email: '',
+  country: '',
+  city: '',
+  file: [],
+  type: ITEM_TYPES[0],
+  category: ITEM_CATEGORY[0],
+};
+
+export default function AddItemForm() {
+  const [item, setItem] = useState(INTIAL_VALUES);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+  };
```

Comment:
Since you used react-hook-form, we will already have the data inside RHF's inbuilt site, so having a local state here is not necessary.